pkgs/applications/networking/cluster/kubectl/default.nix

```diff
@@ -1,7 +1,9 @@
 { stdenv, kubernetes, installShellFiles }:
```

Comment:
might be best to override kubernetes in the callPackage call so that one can pass the full kubernetes if he wants.

Reply:
Hi, I'm  still quite new to nix so I'm not quite sure how to do this. I think I understand the problem in that no matter if a user overrides kubernetes themselves, this code will always override it again. 

My instinct is to change it to `{ stdenv, pkgs, kubernetes ? pkgs.kubernetes.override { components = []; }, installShellFiles }:` but this doesn't do what I want where all the binaries get built rather than just kubectl and kubeadm

Reply:
Look at all-packages.nix: `kubectl = callPackage ../applications/networking/cluster/kubectl { };`
callPackage is a magical function that imports the attributes automatically but you can change the default to be:
`kubectl = callPackage ../applications/networking/cluster/kubectl { kubernetes = kubernetes.override { components = []; }; };`

Reply:
the thing though is that you shouldn't have to compile kubernetes at all it should be in cache.

Reply:
ah all-packages.nix, that's the bit I was missing. 

you are right about the cache but there definitely was a situation where I updated my nix channels, then updated my packages ( via home manager) and I saw all the bits of kubernetes getting recompiled rather than just kubectl. 
So maybe I just happend to update before the new kubernetes version was cached?
Do you think the change is still worthwhile for what might actually be quite  a niche situation?

Reply:
probably sthg wrong with your config. Except if it drastically reduces the size of the closure it 's probably not worth it.

Reply:
doesn't affect the closure size so I'm gonna close the PR